### PR TITLE
swift-reflection-test: Target SwiftRemoteMirror C API

### PR DESCRIPTION
**Pull request by bitjammer**
This tool should test the usage from SwiftRemoteMirror dylib and
the C API, since that is the public interface from which we're
vending the remote reflection functionality.

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
LLDB needs to update its use of readBytes.

**Comment by bitjammer** [comment]
LLDB change: https://github.com/apple/swift-lldb/pull/25

**Comment by rjmccall** [comment]
This doesn't need to change the C++ MemoryReader API; you can just change the C interfaces.

**Comment by bitjammer** [comment]
Fair enough.

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

File: include/swift/Remote/CMemoryReader.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ class CMemoryReader final : public MemoryReader {
 
   bool readBytes(RemoteAddress address, uint8_t *dest, uint64_t size) override {
     return Impl.readBytes(Impl.reader_context,
-                          address.getAddressData(), dest, size);
+                          address.getAddressData(), dest, size) == 1;
```
Comment (rjmccall) [inline include/swift/Remote/CMemoryReader.h]: Prevailing C convention is that you should be testing != 0.

Comment (bitjammer) [inline include/swift/Remote/CMemoryReader.h]: Cool - I'll fix it up before merging.

**Pull request merged**